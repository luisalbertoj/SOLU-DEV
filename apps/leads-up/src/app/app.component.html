<input type="text" [(ngModel)]="message.text" name="message.text" />
<button (click)="sendMessage()">send</button>
<a routerLink="/">Home</a>
<a routerLink="/auth/login">Login</a>
<a routerLink="/auth/register">Register</a>

<router-outlet></router-outlet>
