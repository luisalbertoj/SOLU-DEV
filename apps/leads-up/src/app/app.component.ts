import { Component } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Message } from '@solu-dev/api-interfaces';

@Component({
  selector: 'solu-dev-root',
  templateUrl: './app.component.html',
  styleUrls: ['./app.component.css'],
})
export class AppComponent {
  hello$ = this.http.get<Message>('/api/hello');
  constructor(private http: HttpClient) {}
}
