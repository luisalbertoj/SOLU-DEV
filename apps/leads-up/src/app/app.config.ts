import { Injectable } from '@angular/core';
import { JwtHelperService } from '@auth0/angular-jwt';

const helper = new JwtHelperService();

@Injectable()
export class AppConfig {
  load() {
    return new Promise((resolve) => {
      const token =
        localStorage.getItem('auth_token') === null
          ? false
          : localStorage.getItem('auth_token');
      if (token) {
        try {
          if (helper.isTokenExpired(token) === true) {
            localStorage.removeItem('auth_token');
          }
        } catch (exception) {
          localStorage.removeItem('auth_token');
        }
      }
      resolve(true);
    });
  }
}
