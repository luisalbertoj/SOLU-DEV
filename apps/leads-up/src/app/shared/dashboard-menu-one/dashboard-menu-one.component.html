<div
  class="layout-wrapper"
  (click)="onLayoutClick()"
  [ngClass]="{
    'layout-static': app.menuMode === 'static',
    'layout-overlay': app.menuMode === 'overlay',
    'layout-slim': app.menuMode === 'slim',
    'layout-horizontal': app.menuMode === 'horizontal',
    'layout-sidebar': app.menuMode === 'sidebar',
    'layout-sidebar-static': app.menuMode === 'sidebar' && sidebarStatic,
    'layout-overlay-active': overlayMenuActive,
    'layout-mobile-active': staticMenuMobileActive,
    'layout-static-inactive':
      staticMenuDesktopInactive && app.menuMode === 'static',
    'p-ripple-disabled': !app.ripple
  }"
>
  <div class="layout-main">
    <solu-dev-top-bar-one
      *ngIf="isMobile() || isOverlay() || isHorizontal()"
    ></solu-dev-top-bar-one>

    <solu-dev-menu-one
      *ngIf="!isMobile() && (isStatic() || isSlim() || isSidebar())"
    ></solu-dev-menu-one>

    <solu-dev-breadcrumb-one></solu-dev-breadcrumb-one>

    <div class="layout-main-content">
      <router-outlet></router-outlet>
    </div>

    <solu-dev-footer-one></solu-dev-footer-one>
  </div>

  <solu-dev-right-menu-one></solu-dev-right-menu-one>

  <solu-dev-app-config></solu-dev-app-config>
</div>
