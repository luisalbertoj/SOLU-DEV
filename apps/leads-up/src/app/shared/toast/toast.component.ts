import { CommonModule } from '@angular/common';
import { ChangeDetectionStrategy, Component } from '@angular/core';
import { ToastModule } from 'primeng/toast';
import ToastConfig from './toast.config';

@Component({
  selector: 'solu-dev-toast',
  standalone: true,
  imports: [CommonModule, ToastModule],
  providers: [],
  templateUrl: './toast.component.html',
  styleUrls: ['./toast.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class ToastComponent extends ToastConfig {}
