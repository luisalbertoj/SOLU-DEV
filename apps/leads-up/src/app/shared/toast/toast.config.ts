import { Directive, Input } from '@angular/core';

@Directive()
export default class ToastConfig {
  @Input() protected config!: {
    key: string;
    severity: string;
    summary: string;
    detail: string;
  };
}
