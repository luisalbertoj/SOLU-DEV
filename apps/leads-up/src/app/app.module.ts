import { HashLocationStrategy, LocationStrategy } from '@angular/common';
import { HttpClientModule, HTTP_INTERCEPTORS } from '@angular/common/http';
import { APP_INITIALIZER, NgModule } from '@angular/core';
import { BrowserModule } from '@angular/platform-browser';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
import { TitleStrategy } from '@angular/router';
import { NgxsReduxDevtoolsPluginModule } from '@ngxs/devtools-plugin';
import { NgxsLoggerPluginModule } from '@ngxs/logger-plugin';
import { NgxsModule } from '@ngxs/store';
import { MessageService } from 'primeng/api';
import { environment } from '../environments/environment';
import { AppComponent } from './app.component';
import { AppConfig } from './app.config';
import { AppRoutingModule } from './app.routing';
import { AccessInterceptor } from './core/interceptors/access.interceptor.service';
import { AuthInterceptor } from './core/interceptors/auth.interceptor.service';
import { PageTitleStrategy } from './core/services/page-title.service';
import { TokenProvider } from './core/services/token.service';
import { ToastService } from './shared/toast/services/toast.service';
import { ToastComponent } from './shared/toast/toast.component';
import { MessagesState } from './store/chat/chat.state';

export function servicesOnRun(config: AppConfig, token: TokenProvider) {
  return () => config.load().then(() => token.load());
}
@NgModule({
  declarations: [AppComponent],
  imports: [
    BrowserModule,
    BrowserAnimationsModule,
    AppRoutingModule,
    HttpClientModule,
    NgxsModule.forRoot([], {
      developmentMode: !environment.production,
    }),
    NgxsReduxDevtoolsPluginModule.forRoot({
      disabled: environment.production,
    }),
    NgxsLoggerPluginModule.forRoot({ disabled: environment.production }),
    NgxsModule.forRoot([MessagesState]),
    ToastComponent,
  ],
  providers: [
    AppConfig,
    TokenProvider,
    {
      provide: APP_INITIALIZER,
      useFactory: servicesOnRun,
      multi: true,
      deps: [AppConfig, TokenProvider],
    },
    { provide: HTTP_INTERCEPTORS, useClass: AuthInterceptor, multi: true },
    { provide: HTTP_INTERCEPTORS, useClass: AccessInterceptor, multi: true },
    {
      provide: TitleStrategy,
      useClass: PageTitleStrategy,
    },
    {provide: LocationStrategy, useClass: HashLocationStrategy},
    ToastService,
    MessageService,
  ],
  bootstrap: [AppComponent],
})
export class AppModule {}
