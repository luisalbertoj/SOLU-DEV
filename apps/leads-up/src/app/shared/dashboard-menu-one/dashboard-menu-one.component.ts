import { CommonModule } from '@angular/common';
import { Component } from '@angular/core';
import { RouterModule } from '@angular/router';
import { PrimeNGConfig } from 'primeng/api';
import { Subject } from 'rxjs';
import { AppComponent } from '../../app.component';
import { AppConfigComponent } from '../app-config/app-config.component';
import { BreadcrumbOneComponent } from '../breadcrumb-one/breadcrumb-one.component';
import { FooterOneComponent } from '../footer-one/footer-one.component';
import { MenuOneComponent } from '../menu-one/menu-one.component';
import { RightMenuOneComponent } from '../right-menu-one/right-menu-one.component';
import { TopBarOneComponent } from '../top-bar-one/top-bar-one.component';
import { MenuService } from './services/dashboard-menu.service';

@Component({
  selector: 'solu-dev-dashboard-menu-one',
  standalone: true,
  imports: [
    CommonModule,
    TopBarOneComponent,
    MenuOneComponent,
    BreadcrumbOneComponent,
    RouterModule,
    AppConfigComponent,
    FooterOneComponent,
    RightMenuOneComponent,
  ],
  providers: [MenuService],
  templateUrl: './dashboard-menu-one.component.html',
  styleUrls: ['./dashboard-menu-one.component.scss'],
})
export class DashboardMenuOneComponent {
  overlayMenuActive!: boolean;

  staticMenuDesktopInactive = false;

  staticMenuMobileActive!: boolean;

  sidebarActive$ = new Subject<boolean>();

  sidebarActive = false;

  sidebarStatic = false;

  menuClick!: boolean;

  menuHoverActive = false;

  topbarMenuActive!: boolean;

  topbarItemClick!: boolean;

  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  activeTopbarItem!: any;

  configActive$ = new Subject<boolean>();

  configActive!: boolean;

  configClick!: boolean;

  rightMenuActive$ = new Subject<boolean>();

  rightMenuActive!: boolean;

  rightMenuClick!: boolean;

  searchActive!: boolean;

  searchClick!: boolean;

  activeInlineProfile!: boolean;

  pinActive!: boolean;

  constructor(
    private menuService: MenuService,
    private primengConfig: PrimeNGConfig,
    public app: AppComponent
  ) {}

  onLayoutClick() {
    if (!this.topbarItemClick) {
      this.activeTopbarItem = null;
      this.topbarMenuActive = false;
    }

    if (this.configActive && !this.configClick) {
      this.configActive = false;
      this.configActive$.next(false);
    }

    if (this.rightMenuActive && !this.rightMenuClick) {
      this.rightMenuActive$.next(false);
      this.rightMenuActive = false;
    }

    if (this.searchActive && !this.searchClick) {
      this.searchActive = false;
    }

    if (!this.menuClick) {
      if ((this.isSlim() || this.isHorizontal()) && !this.isMobile()) {
        this.menuService.reset();
        this.menuHoverActive = false;
      }

      if (this.overlayMenuActive || this.staticMenuMobileActive) {
        this.overlayMenuActive = false;
        this.staticMenuMobileActive = false;
      }
    }

    this.configClick = false;
    this.rightMenuClick = false;
    this.searchClick = false;
    this.menuClick = false;
    this.topbarItemClick = false;
  }

  onSidebarClick() {
    this.menuClick = true;
  }

  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  onToggleMenu(event: any) {
    this.menuClick = true;

    if (this.overlayMenuActive) {
      this.overlayMenuActive = false;
    }

    if (this.sidebarActive) {
      this.sidebarStatic = !this.sidebarStatic;
    }

    event.preventDefault();
  }

  onSidebarMouseOver() {
    if (this.app.menuMode === 'sidebar' && !this.sidebarStatic) {
      this.sidebarActive$.next(this.isDesktop());
      this.sidebarActive = this.isDesktop();
      setTimeout(() => {
        this.pinActive = this.isDesktop();
      }, 200);
    }
  }

  onSidebarMouseLeave() {
    if (this.app.menuMode === 'sidebar' && !this.sidebarStatic) {
      setTimeout(() => {
        this.sidebarActive$.next(false);
        this.sidebarActive = false;
        this.pinActive = false;
      }, 250);
    }
  }
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  onMenuButtonClick(event: any) {
    this.menuClick = true;

    if (this.isOverlay()) {
      this.overlayMenuActive = !this.overlayMenuActive;
    }

    if (this.isDesktop()) {
      this.staticMenuDesktopInactive = !this.staticMenuDesktopInactive;
    } else {
      this.staticMenuMobileActive = !this.staticMenuMobileActive;
    }

    event.preventDefault();
  }
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  onTopbarItemClick(event: any, item: any) {
    this.topbarItemClick = true;

    if (this.activeTopbarItem === item) {
      this.activeTopbarItem = null;
    } else {
      this.activeTopbarItem = item;
    }

    event.preventDefault();
  }
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  onTopbarSubItemClick(event: any) {
    event.preventDefault();
  }
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  onRippleChange(event: any) {
    this.app.ripple = event.checked;
    this.primengConfig.ripple = event.checked;
  }

  onConfigClick() {
    this.configClick = true;
  }

  onRightMenuButtonClick() {
    this.rightMenuClick = true;
    this.rightMenuActive = true;
    this.rightMenuActive$.next(true);
  }

  onRightMenuClick() {
    this.rightMenuClick = true;
  }

  isStatic() {
    return this.app.menuMode === 'static';
  }

  isOverlay() {
    return this.app.menuMode === 'overlay';
  }

  isSlim() {
    return this.app.menuMode === 'slim';
  }

  isHorizontal() {
    return this.app.menuMode === 'horizontal';
  }

  isSidebar() {
    return this.app.menuMode === 'sidebar';
  }

  isDesktop() {
    return window.innerWidth > 991;
  }

  isMobile() {
    return window.innerWidth <= 991;
  }
}
