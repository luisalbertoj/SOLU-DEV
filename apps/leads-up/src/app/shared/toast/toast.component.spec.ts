import { ComponentFixture, TestBed } from '@angular/core/testing';
import { NoopAnimationsModule } from '@angular/platform-browser/animations';
import { RouterTestingModule } from '@angular/router/testing';
import { MessageService } from 'primeng/api';
import { AppComponent } from '../../app.component';
import { DashboardMenuOneComponent } from '../dashboard-menu-one/dashboard-menu-one.component';
import { MenuService } from '../dashboard-menu-one/services/dashboard-menu.service';

import { ToastComponent } from './toast.component';

describe('ToastComponent', () => {
  let component: ToastComponent;
  let fixture: ComponentFixture<ToastComponent>;
  let compiled: HTMLElement;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [],
      imports: [RouterTestingModule, NoopAnimationsModule],
      providers: [
        AppComponent,
        DashboardMenuOneComponent,
        MenuService,
        MessageService,
      ],
    }).compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(ToastComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
    compiled = fixture.nativeElement;
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('this should be equal to the snapshot', () => {
    expect(compiled.innerHTML).toMatchSnapshot();
  });
});
