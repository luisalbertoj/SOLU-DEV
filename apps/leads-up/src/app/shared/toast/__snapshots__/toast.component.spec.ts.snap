// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToastComponent this should be equal to the snapshot 1`] = `"<p-toast class=\\"p-element ng-tns-c61-0\\"><div class=\\"ng-tns-c61-0 p-toast p-component p-toast-top-right\\" ng-reflect-ng-class=\\"p-toast p-component p-toast-to\\"><!--bindings={}--></div></p-toast>"`;
